server/neptune/gateway/event/pull_request_handler.go

```diff
@@ -20,84 +44,113 @@ type PullRequest struct {
 	InstallationToken int64
 }
 
-func NewAutoplannerValidatorProxy(
-	autoplanValidator Validator,
-	logger logging.Logger,
-	workerProxy *PullEventWorkerProxy,
-	scheduler scheduler,
-) *AutoplannerValidatorProxy {
-	return &AutoplannerValidatorProxy{
-		autoplanValidator: autoplanValidator,
-		workerProxy:       workerProxy,
-		logger:            logger,
-		scheduler:         scheduler,
+func NewModifiedPullHandler(logger logging.Logger, workerProxy *PullSNSWorkerProxy, scheduler scheduler, rootConfigBuilder rootConfigBuilder, globalCfg valid.GlobalCfg, updater vcsStatusUpdater, requirementChecker requirementChecker) *ModifiedPullHandler {
+	return &ModifiedPullHandler{
+		WorkerProxy:        workerProxy,
+		Logger:             logger,
+		Scheduler:          scheduler,
+		RootConfigBuilder:  rootConfigBuilder,
+		GlobalCfg:          globalCfg,
+		VCSStatusUpdater:   updater,
+		RequirementChecker: requirementChecker,
 	}
 }
 
-type Validator interface {
-	InstrumentedIsValid(ctx context.Context, logger logging.Logger, baseRepo models.Repo, headRepo models.Repo, pull models.PullRequest, user models.User) bool
-}
-
-type Writer interface {
-	WriteWithContext(ctx context.Context, payload []byte) error
-}
-
-type AutoplannerValidatorProxy struct {
-	autoplanValidator Validator
-	workerProxy       *PullEventWorkerProxy
-	logger            logging.Logger
-	scheduler         scheduler
+func (p *ModifiedPullHandler) Handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
+	return p.Scheduler.Schedule(ctx, func(ctx context.Context) error {
+		return p.handle(ctx, request, event)
+	})
 }
 
-func (p *AutoplannerValidatorProxy) handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
-	if ok := p.autoplanValidator.InstrumentedIsValid(
-		ctx,
-		p.logger,
-		event.Pull.BaseRepo,
-		event.Pull.HeadRepo,
-		event.Pull,
-		event.User); ok {
-		return p.workerProxy.Handle(ctx, request, event)
+func (p *ModifiedPullHandler) handle(ctx context.Context, request *http.BufferedRequest, event PullRequest) error {
+	criteria := requirement.Criteria{
+		User:              event.User,
+		Branch:            event.Pull.HeadBranch,
+		Repo:              event.Pull.HeadRepo,
+		OptionalPull:      &event.Pull,
+		InstallationToken: event.InstallationToken,
+	}
+	if err := p.RequirementChecker.Check(ctx, criteria); err != nil {
+		return errors.Wrap(err, "checking pr requirements")
 	}
 
-	p.logger.WarnContext(ctx, "request isn't valid and will not be proxied to sns")
+	commit := &config.RepoCommit{
+		Repo:          event.Pull.HeadRepo,
+		Branch:        event.Pull.HeadBranch,
+		Sha:           event.Pull.HeadCommit,
+		OptionalPRNum: event.Pull.Num,
+	}
 
-	return nil
-}
+	// set clone depth to 1 for repos with a branch checkout strategy
```

Comment:
Why do we do this again? Can you elaborate on the comment

Reply:
We have a checkout strategy configuration to support shallow cloning for large repos: https://github.com/lyft/atlantis/blob/release-v0.17.3-lyft.1/server/events/working_dir.go#L240. I can explain in comment

Reply:
Just to add, we will no longer need to perform merges with the main branch since we are requiring rebases so the name for this configuration is a bit dated, but it's still valuable in reducing the amount of time it takes to clone larger repos.